tools/sklearn/to_categorical.xml

```diff
@@ -0,0 +1,93 @@
+    <expand macro="macro_stdio"/>
+    <version_command>echo "@VERSION@"</version_command>
+    <command detect_errors="exit_code"><![CDATA[
+        python '$__tool_directory__/to_categorical.py'
+            --inputs '$inputs'
+            --infile '$infile'
+            #if $num_classes
+            --num_classes '$num_classes'
+            #end if
+            --outfile '$outfile'
+    ]]>
+    </command>
+    <configfiles>
+        <inputs name="inputs" />
+    </configfiles>
+    <inputs>
+        <param name="infile" type="data" format="tabular" label="Input file"/>
+        <param name="header0" type="boolean" optional="true" truevalue="booltrue" falsevalue="boolfalse" checked="true" label="Does the dataset contain header?"/>
+        <param name="num_classes" type="integer" optional="true" label="Total number of classes"/>
+    </inputs>
+    <outputs>
+        <data name="outfile" format="tabular"/>
+    </outputs>
+    <tests>
+        <test>
+            <param name="infile" value="ohe_in_w_header.tabular" ftype="tabular"/>
+            <param name="header0" value="true"/>
+            <output name="outfile" file="ohe_out_4.tabular" ftype="tabular"/>
+        </test>
+        <test>
+            <param name="infile" value="ohe_in_w_header.tabular" ftype="tabular"/>
+            <param name="header0" value="true"/>
+            <param name="num_classes" value="4" />
+            <output name="outfile" file="ohe_out_4.tabular" ftype="tabular"/>
+        </test>
+        <test>
+            <param name="infile" value="ohe_in_w_header.tabular" ftype="tabular"/>
+            <param name="header0" value="true"/>
+            <param name="num_classes" value="5" />
+            <output name="outfile" file="ohe_out_5.tabular" ftype="tabular"/>
+        </test>
+        <test>
+            <param name="infile" value="ohe_in_wo_header.tabular" ftype="tabular"/>
+            <param name="header0" value="false"/>
+            <output name="outfile" file="ohe_out_4.tabular" ftype="tabular"/>
+        </test>
+        <test>
+            <param name="infile" value="ohe_in_wo_header.tabular" ftype="tabular"/>
+            <param name="header0" value="false"/>
+            <param name="num_classes" value="4" />
+            <output name="outfile" file="ohe_out_4.tabular" ftype="tabular"/>
+        </test>
+        <test>
+            <param name="infile" value="ohe_in_wo_header.tabular" ftype="tabular"/>
+            <param name="header0" value="false"/>
+            <param name="num_classes" value="5" />
+            <output name="outfile" file="ohe_out_5.tabular" ftype="tabular"/>
+        </test>
+    </tests>
+    <help><![CDATA[
+**What it does**
+
+Converts a class vector (integers) to binary class matrix.
+
+tf.keras.utils.to_categorical(
+    y, num_classes=None, dtype='float32'
+)
+
+E.g. for use with categorical_crossentropy.
+
+Arguments
+
+y: a vector of numbers to be converted into a vector of one-hot encoded values.
```

Comment:
the transformation should be a matrix, correct?

Reply:
Yes, will revise.